src/composables/alarmModel.ts

```diff
@@ -1,98 +1,109 @@
-import { Storage } from '@capacitor/storage'
+import { Preferences } from '@capacitor/preferences'
 // import alarmsJson from '../alarms.json'
-import routesJson from '../routes.json'
-import stationsJson from '../stations.json'
-import tagsJson from '../tags.json'
-import transportJson from '../transport.json'
-import iconsJson from '../icons.json'
-import backgroundsJson from '../backgrounds.json'
+import routesJson from "../routes.json";
+import stationsJson from "../stations.json";
+import tagsJson from "../tags.json";
+import transportJson from "../transport.json";
+import iconsJson from "../icons.json";
+import backgroundsJson from "../backgrounds.json";
 
 class Alarm {
-  id: number
-  enabled: boolean
-  tags: Tag[]
-  settings: Settings
-  expiration: number
+  id: number;
+  enabled: boolean;
+  tags: Tag[];
+  settings: Settings;
+  expiration: number;
 
   constructor(id?: number) {
     if (id !== undefined) {
-      this.getAlarm(id)
-    }
-    else {
-      this.id = -1
-      this.enabled = true
-      this.tags = [new Tag()]
-      this.settings = new Settings()
-      this.expiration = 0
+      this.getAlarm(id);
+    } else {
+      this.id = -1;
+      this.enabled = true;
+      this.tags = [new Tag()];
+      this.settings = new Settings();
+      this.expiration = 0;
     }
   }
 
   async getAlarm(id: number): Promise<Alarm> {
-    if (id === -1) return new Alarm()
-    const alrm = await Storage.get({ key: 'alarms' })
-      .then((alarms) => {
-        // if (!alarms.value) throw new Error('No alarms found')
-        if (!alarms.value) return undefined
-        // console.log(alarms)
-        const alarm = JSON.parse(alarms.value).find((alarm) => { return alarm.id === id })
-        // JSON.parse(JSON.stringify()) is used to clone the object, used as a fast fix of vue reactvity
-        // TODO: fix it
-        //  || (alarm = JSON.parse(JSON.stringify(alarmsJson[id])))
-        if (alarm) {
-          this.id = id
-          this.enabled = alarm.enabled
-          // console.log(alarm)
-          this.tags = alarm.tags.map(tag => new Tag(tag))
-          alarm.settings.general.route = new Route(alarm.settings.general.route)
-          alarm.settings.general.station = new Station(alarm.settings.general.station)
-          alarm.settings.general.image.ico = new Icon(alarm.settings.general.image.ico)
-          alarm.settings.general.image.bg = new Background(alarm.settings.general.image.bg)
-          this.settings = new Settings(alarm.settings)
-          this.expiration = alarm.expiration
-        }
-        // JSON.parse(JSON.stringify()) is used to clone the object, used as a fast fix of vue reactvity
-        // TODO: fix it
-        return JSON.parse(JSON.stringify(this))
-      })
-    return alrm
+    if (id === -1) return new Alarm();
+    
+    // TODO: Why do we need await here if you used Promise chains?
+    const alrm = await Preferences.get({ key: "alarms" }).then((alarms) => {
```

Comment:
```suggestion
    const alarms = await Preferences.get({ key: "alarms" })
    // use alarms like in callback
```

__Also check:__
[Preferences API](https://capacitorjs.com/docs/apis/preferences)